### ISSUE TITLE
Problems using new maven "build-image" goal in corporate environment with private docker registry; in docker toolbox is also problematic to run it (2.3.0.M2)

### ISSUE THREAD

**Issue by artem-korolev** [spring-projects/spring-boot#20289]
1. First (and more important)

When I run "build-image" goal in windows with Docker Toolbox,, I'm getting this error in log:

```
[INFO] I/O exception (java.io.FileNotFoundException) caught when processing request to {}->docker://localhost:2376: \\.\pipe\docker_engine (The system cannot find the file specified)
```

I found this is hardcoded here, if I got it right:
https://github.com/spring-projects/spring-boot/blob/e28338d6cd483896dc1632653cff90d5da8fd9e9/spring-boot-project/spring-boot-tools/spring-boot-buildpack-platform/src/main/java/org/springframework/boot/buildpack/platform/docker/DockerApi.java#L96

will be good to have this url configurable in spring-boot-maven-plugin, or/and taken from "DOCKER_HOST" environment variable. IMHO its better to take it from environment variable, then i don't need profiles for different systems. So yes. I think DOCKER_HOST environment is better option here

**Comment by snicoll** [comment]
@artem-korolev thanks for testing the milestone.

Please refrain from reporting two separate problems in the same issue as it makes it very hard to track things. The build plugin uses the local docker daemon which is available out of the box with Docker Desktop. If you are running Docker Toolbox, please see [this comment](https://github.com/spring-projects/spring-boot/issues/19941#issuecomment-578714760) to configure it. 

Let's focus on your first issue. Can you please run that build with `--stacktrace` and share the log?

**Comment by artem-korolev** [comment]
Well.. I just figured out seems like the core of first issue. I'm using original cloudfoundry/cnb image, which have instructions forwarding to docker.io
I just need to build my own builder and point spring boot to use it instead of original one.

Regarding second problem - I will check comment you suggested. Hope it helps.

Closing this issue to avoid the noise. Thanks you for such a fast answer!
